src/middlewared/middlewared/plugins/cron.py

```diff
@@ -16,6 +16,20 @@ def cron_extend(self, data):
         Cron.convert_db_format_to_schedule(data)
         return data
 
+    @private
+    async def construct_cron_command(self, schedule, user, command, stdout=True, stderr=True):
+        return list(
+            filter(
```

Comment:
What is this filter for?

Reply:
So when we join this list on space, if `stdout` is `''`, it adds an extra space in the final command. This filter flushes `stdout/stderr` if they are emtpy